src/language-css/printer-postcss.js

```diff
@@ -189,6 +202,50 @@ function genericPrint(path, options, print) {
     }
     case "css-atrule": {
       const parentNode = path.getParentNode();
+      const isTemplatePlaceholderNodeWithoutSemiColon =
+        isTemplatePlaceholderNode(node) &&
+        !parentNode.raws.semicolon &&
+        options.originalText[options.locEnd(node) - 1] !== ";";
+
+      if (node.mixin) {
+        return concat([
+          path.call(print, "selector"),
+          node.important ? " !important" : "",
+          isTemplatePlaceholderNodeWithoutSemiColon ? "" : ";"
+        ]);
+      }
+
+      if (node.function) {
+        return concat([
+          node.name,
+          concat([path.call(print, "params")]),
+          isTemplatePlaceholderNodeWithoutSemiColon ? "" : ";"
+        ]);
+      }
+
+      if (node.variable) {
+        return concat([
+          "@",
+          node.name,
+          ": ",
+          node.value ? concat([path.call(print, "value")]) : "",
+          node.raws.between.trim() ? node.raws.between.trim() + " " : "",
+          node.nodes
+            ? concat([
+                "{",
+                indent(
+                  concat([
+                    node.nodes.length > 0 ? softline : "",
+                    printNodeSequence(path, options, print)
+                  ])
+                ),
+                softline,
+                "}"
+              ])
+            : "",
+          isTemplatePlaceholderNodeWithoutSemiColon ? "" : ";"
+        ]);
+      }
```

Comment:
Each `node.mixin`/`node.function`/`node.variable` should be:
```
if (isLessParser) { 
  if (node.mixin) { /* ... */ }
  if (node.function) { /* ... */ }
}
```

It is better for performance and other parsers can potentially have same properties

Reply:
will do

Reply:
Please add `isLessParser` check here

Reply:
In fact we can't use `options.parser === "less"`, since we has `parser=css` also use `less` parser, so I use `options.parser !== "scss"` in this commit

https://github.com/prettier/prettier/pull/6981/commits/07010de356026f1568864bfceb14c271a39e8088

What do you think? or maybe `options.parser === "less" || options.parser === "csss"`?

Reply:
To be honestly we should throw error on less syntax when developer set `css` parser. What about change the `options.parser` value change when we switch on other syntax?

Reply:
Maybe we can do this later in a seperate PR, shall we?

Reply:
Yes, don't forget open new issue after merge, anyway let's encapsulate logic in one util function `isLessParser`, it is allow to fix problem in one places